src/vodozemac/pk_encryption.rs

```diff
@@ -0,0 +1,183 @@
+//! This module provides public-key encryption and decryption functionality
+//! using the `vodozemac` cryptographic library.
+//!
+//! This implements the m.megolm_backup.v1.curve25519-aes-sha2 algorithm
+//! described in the Matrix [spec].
+//!
+//! More details can be found in the official [vodozemac documentation].
+//!
+//! [spec]: https://spec.matrix.org/v1.11/client-server-api/#backup-algorithm-mmegolm_backupv1curve25519-aes-sha2
+//! [vodozemac documentation]: https://docs.rs/vodozemac/latest/vodozemac/pk_encryption/
+
+#![allow(missing_debug_implementations)]
+
+use matrix_sdk_crypto::vodozemac::{base64_encode, pk_encryption};
+use wasm_bindgen::prelude::*;
+
+use super::{Curve25519PublicKey, Curve25519SecretKey};
+
+/// A class representing an encrypted message using `PkEncryption`.
```

Comment:
it would be nice to link all the types together, to make navigating the docs easier.

```suggestion
/// A class representing an encrypted message using {@link PkEncryption}.
```

Likewise all the other references in this file.

Reply:
Added links in 519952bf8e9320a184dc23834774eed6622bb873. 

Not sure if you'd like to have links to `PkEncryption` when we mention `PkEncryption` in a method for `PkEncryption` itself? I left those out, but happy to add them if you prefer that as well.

Reply:
> Not sure if you'd like to have links to `PkEncryption` when we mention `PkEncryption` in a method for `PkEncryption` itself? I left those out, but happy to add them if you prefer that as well.

I don't think that's necessary, but thanks